lib/utils.bash

```diff
@@ -806,3 +806,20 @@ with_shim_executable() {
 
   return 126
 }
+
+substitute() {
+  # Use Bash substituion rather than sed as it will handle escaping of all
+  # strings for us.
+  local input=$1
+  local find_str=$2
+  local replace=$3
+  printf "%s" "${input//"$find_str"/"$replace"}"
+}
```

Comment:
A bit more code now, but this is much more robust than `sed` because it can handle any character in the three input strings and will handle escaping for us.

Reply:
Is this Bash specific?

Reply:
Yes I believe so. But this is a Bash file and should only be executed by Bash.